### PR TITLE
Make checkpoint_files = get_all_checkpoints() by default

### PR DESCRIPTION
**Pull request by benclifford**
This is the sane default for this parameter in almost all
checkpointing situations that I've encountered.

The previous behaviour of loading no checkpoint files, but
checkpointing new activity can be obtained by writing:

checkpoint_files = []

This will be a breaking change for anyone expecting the previous default behaviour.

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Comment by yadudoc** [comment]
The changes look right, but I want to give others a chance to weigh in on whether it makes sense to have this be the default behavior. I worry that use-cases that have apps that depend on libraries in development will get silently skipped.

**Review by yadudoc — DISMISSED** [review]
The changes look right, and I'm approving this PR. After a brief discussion over on slack, it doesn't look like there are any complaints about the change to the default behavior. 

Since @annawoodard and @ravihansa3000 have more experience with checkpointing in practice, I'm hoping they'd chime in soon, if they have any reservations against this change.

@benclifford it looks like there's a merge conflict that needs fixing.

File: parsl/dataflow/dflow.py [hunk 1]
```diff
@@ -152,7 +152,13 @@ def __init__(self, config=Config()):
             self.monitoring.send(MessageType.WORKFLOW_INFO,
                                  workflow_info)
 
-        checkpoints = self.load_checkpoints(config.checkpoint_files)
+        if config.checkpoint_files is not None:
+            checkpoints = self.load_checkpoints(config.checkpoint_files)
+        elif config.checkpoint_files is None and config.checkpoint_mode is not None:
+            checkpoints = self.load_checkpoints(get_all_checkpoints(self.run_dir))
+        else:
+            checkpoints = {}
```
Comment (ravihansa3000) [inline parsl/dataflow/dflow.py:160]: shouldn't this be `checkpoints = []` ?

Comment (benclifford) [reply parsl/dataflow/dflow.py:160]: `load_checkpoints` returns a dict (of arguments->result mappings) - what this is defaulting to empty is that dict, not defaulting the list of what to load.

Comment (yadudoc) [reply parsl/dataflow/dflow.py:160]: This is correct, `load_checkpoints` returns the memoized task table.

**Review by yadudoc — APPROVED** [review]

**Pull request merged**